specfile/specfile.py

```diff
@@ -179,11 +256,16 @@ def rpm_spec(self) -> rpm.spec:
 
     def reload(self) -> None:
         """Reloads the spec file content."""
-        self._lines, self._trailing_newline = self._read_lines(self.path)
+        self._lines, self._trailing_newline = self._read_lines(self._file)
 
     def save(self) -> None:
         """Saves the spec file content."""
-        self.path.write_text(str(self), encoding="utf8", errors="surrogateescape")
+        if not self.is_writable:
+            raise OSError("This stream is read-only")
```

Comment:
I think this is unnecessary, a (perhaps more useful) exception would be raised anyway.